mountpoint-s3/tests/fuse_tests/semantics_doc_test.rs

```diff
@@ -0,0 +1,158 @@
+use std::path::Path;
+
+use fuser::BackgroundSession;
+use mountpoint_s3::S3FilesystemConfig;
+use tempfile::TempDir;
+use walkdir::WalkDir;
+
+use crate::fuse_tests::TestClientBox;
+
+/// Recursively list the contents of a directory and return the paths of all entries, with the
+/// initial `path` stripped. If `files` is true, the list contains only files; if false, it contains
+/// only directories.
+fn list_dir_recursive(path: impl AsRef<Path>, files: bool) -> Result<Vec<String>, walkdir::Error> {
+    let path_prefix = format!("{}/", path.as_ref().to_string_lossy());
+    let contents_iter = WalkDir::new(path)
+        .sort_by_file_name()
+        .into_iter()
+        .collect::<Result<Vec<_>, _>>()?;
+    let contents_filtered = contents_iter.iter().filter(|d| {
```

Comment:
Is not this filtering the obvious? What are we trying to filter out here?

Reply:
We want to only include files, or only include directories, because there's no way to tell which is which just from their name.